src/components/search.js

```diff
@@ -0,0 +1,79 @@
+/* eslint-disable comma-dangle */
+/* eslint-disable object-curly-newline */
+import IconSvg from '../assets/icons.svg';
+import API from '../controller/api';
+import mainPage from '../pages/background';
+import dataComponent from './dataComp';
+import cityNotFoundComponent from './city-not-found';
+
+const searchComponent = (() => {
+  const searchInputRender = () => `
+  <div class="input">
+    <div class="input-container">
+      <input type="text" id="search-input" placeholder="City" autocomplete="off">
+      <svg class="svg-icon" id="search-button">
+        <use href="${IconSvg}#search-symbol"></use>
+      </svg>
+    </div>
+  </div>
+`;
+  const renderPageAfterSearch = async (mainElement, searchInput) => {
+    const cityPage = document.querySelector('.city-not-found');
+    const cardContainer = document.querySelector('.card-container');
+    try {
+      const returnCityData = await Promise.all([
+        API.weatherDataByCity(searchInput.value),
+        API.cityBackgroundImage(searchInput.value),
+      ]);
+
+      const { main, weather, wind, clouds, sys } = await returnCityData[0];
```

Comment:
- [ ] [OPTIONAL] When the returnCityData doesn't return these fields, for example when it has an error related to the API key, you could add a try/catch block to show a message to the user like 'Something went wrong' instead of having the app crash.

Reply:
Yes, I have to catch that eror